Ascend-Server.api/Controllers/v1/GuidedJournalEntryController.cs

```diff
@@ -0,0 +1,166 @@
+using Data.Exceptions;
+using Microsoft.AspNetCore.Mvc;
+using AutoMapper;
+using Dto;
+using IServices;
+using Ascend_Server.api.ActionFilters;
+using Microsoft.AspNetCore.Authorization;
+using Services;
+
+namespace Controllers;
+
+[Authorize]
+[ApiController]
+[Route("api/v{version:apiVersion}/[controller]")]
+[ServiceFilter(typeof(ModelStateActionFilter))]
+[ApiVersion("1.0")]
+public class GuidedJournalEntryController : ControllerBase
+{
+    private readonly IGuidedJournalEntryService _guidedJournalEntryService;
+
+    private readonly IMapper _mapper;
+
+    public GuidedJournalEntryController(IGuidedJournalEntryService guidedJournalEntryService,
+                                        IMapper mapper)
+    {
+        _guidedJournalEntryService = guidedJournalEntryService;
+
+        _mapper = mapper;
+    }
+    /// <summary>
+    /// Retrieves a single Guided Journal Entry that matches the passed id.
+    /// </summary>
+    /// <returns>
+    /// Returns a single instance of type <see cref="Dto.GuidedJournalEntry"/> object matching the id passed.
+    /// </returns>
+    /// <response code="200">Returns the Guided Journal Entry.</response>
+    /// <response code="400">The request was invalid.</response>
+    /// <response code="404">No guided journal entry found for id.</response>
+    /// 
+    [HttpGet("id/{id}")]
+    public ActionResult<Dto.GuidedJournalEntry> GetById(Guid id)
+    {
+        try
+        {
+            var guidedJournalEntry = _guidedJournalEntryService.GetById(id);
+
+            var dto = _mapper.Map<Data.GuidedJournalEntry, Dto.GuidedJournalEntry>(guidedJournalEntry);
+
+            return new OkObjectResult(dto);
+        }
+        catch (NotFoundException e)
+        {
+            return NotFound(e.Message);
+        }
+
+    }
+    /// <summary>
+    /// Gets all guided journal entries for the authenticated user.
+    /// </summary>
+    /// <returns>A list of GuidedJournalEntry DTOs if successful, or a NotFoundResult if no entries were found.</returns>
+    /// <response code="200">Returns a list of GuidedJournalEntry DTOs if successful.</response>
+    /// <response code="404">If no entries were found for the authenticated user.</response>
+    [HttpGet("userId/{userId}")]
+    public IActionResult GetAll(Guid userId)
+    {
+        try
+        {
+            var guidedJournalEntries = _guidedJournalEntryService.GetAllForUserId(userId);
+
+            if (guidedJournalEntries == null)
+            {
+                return NotFound();
+            }
+
+            var dtos = _mapper.Map<Data.GuidedJournalEntry[], Dto.GuidedJournalEntry[]>(guidedJournalEntries);
+
+            return new OkObjectResult(dtos);
+        }
+        catch (Exception)
+        {
+            return new BadRequestResult();
```

Comment:
bad request means that the user can change the request to correct the problem. We don't know that to be true on a general exception. You could catch other exceptions and return a bad request but you should probably be returning an internal server error on general exceptions. Follow this practice in all controller actions.